### DISCUSSION TITLE
Apple M1 cfclient Installation Issue – Error on Startup with Quartz Framework

### DISCUSSION THREAD

**Discussion by philwa9** [bitcraze/discussions#1100] [Q&A]
Hi everyone,

I have tried to install and open the cfclient on my Apple M1. However, when I try to start the client with the command "cfclient" I get this error:

`OSError: dlopen(Quartz.framework/Quartz, 0x0006): tried: 'Quartz.framework/Quartz' (no such file), '/System/Volumes/Preboot/Cryptexes/OSQuartz.framework/Quartz' (no such file), '/usr/local/lib/Quartz.framework/Quartz' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/usr/local/lib/Quartz.framework/Quartz' (no such file), '/usr/lib/Quartz.framework/Quartz' (no such file, not in dyld cache), 'Quartz.framework/Quartz' (no such file)`

Quartz is installed in /System/Library/Frameworks/Quartz.framework.

I tried installing the client both ways: using the arch --x86_64 command for the arm64 architecture and the normal installation for x86 (which worked fine on a friend's M2 MacBook).

Has anyone encountered this problem and has a solution?

**Comment by ataffanel** [comment]
Hi,
I assume you are following the instruction with brew (https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/installation/install/#mac).

This is not something we have experienced before, searching quickly for the error I found some hit on Stack-overflow so you may be able to find a solution on the web. (for example https://stackoverflow.com/questions/42530309/no-such-file-requirements-txt-error-while-installing-quartz-module?noredirect=1).

If you find a solution please post it, otherwise answer that you did not and I will be able to investigate more early January when I have access to the office's dev M1 mac. To help me reproduce the issue can you also tell me what OS you are running.

**Reply by philwa9** [reply]
Thanks for your reply!
Yes, that's right. I was referring to the introduction you posted.

I managed to install quartz using the stack overflow fix. However, I still get the same problem when I try to run cfclient. My Mac is running MacOS Sonoma 14.2.1.

**Reply by knmcguire** [reply]
Hi! Sorry for the late replies, many of us are on holidays and we can't try it out on a M1 for you at the moment... 

Are you still having this problem or have you found a solution in the mean time already? If not than we are back in the office next week and we can help you then.

**Reply by r-bt** [reply]
I am also getting this issue. I installed python using pyenv which I installed from x86 brew. I'm on macOS 13.5.2.

**Reply by r-bt** [reply]
It looks like it's an issue with the vispy library. My quick fix right now is changing:
`quartz = cdll.LoadLibrary(util.find_library('quartz'))`

to 
`quartz = cdll.LoadLibrary('/System/Library/Frameworks/Quartz.framework/Quartz')`

in `vispy/ext/cocoapy.py`

**Reply by knmcguire** [reply]
This is awesome! I've just tried it out myself and it works again. Thank you for this!

It's a bit of dirty fix but at least it is a workable situation for people that want to run the cfclient from an m1

**Reply by knmcguire** [reply]
I'll mention it in this open ticket for it: https://github.com/bitcraze/crazyflie-clients-python/issues/694
